runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -695,8 +695,10 @@ public void evaluate(Reshuffle<K, V> transform, EvaluationContext context) {
         final WindowedValue.WindowedValueCoder<KV<K, V>> wvCoder =
             WindowedValue.FullWindowedValueCoder.of(coder, windowFn.windowCoder());
 
+        int numPartitions =
+            Math.max(context.getSparkContext().defaultParallelism(), inRDD.getNumPartitions());
```

Comment:
I believe we should do the same thing in the streaming and batch portable pipeline translators as well.